thermostat-card/thermostat.js

```diff
@@ -0,0 +1,208 @@
+import thermostatDial from '/local/custom-lovelace/thermostat-card/thermostat_lib.js?module';
+
+class ThermostatCard extends HTMLElement  {
+  static get properties() {
+    return {
+      _hass: Object,
+      config: Object,
+    }
+  }
+
+  constructor() {
+    super();
+    this.attachShadow({ mode: 'open' });
+  }
+
+  setConfig(config) {
+    if (!config.entity) {
+      throw new Error('Please define an entity');
+    }
+
+    const root = this.shadowRoot;
+    if (root.lastChild) root.removeChild(root.lastChild);
+
+    const cardConfig = Object.assign({}, config);
+
+    const card = document.createElement('ha-card');
+    const shadow = card.attachShadow({ mode: 'open' });
+    const content = document.createElement('div');
+    const style = document.createElement('style');
+    style.textContent = `
+    @import url(https://fonts.googleapis.com/css?family=Open+Sans:300);
+    body {
+      background-color: #CCCCCC;
+    }
+    #thermostat {
+      width: 50vmin;
+      height: 50vmin;
+      margin: 0 auto;
+      -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
```

Comment:
Please avoid using custom extension that only works on some browsers. We already have some exceptions because shadowdom doesn't work on Edge and it works only on Firefox nightly (hope released soon on stable).